packages/gatsby/src/redux/types.ts

```diff
@@ -320,6 +320,8 @@ export interface IGatsbyState {
   }
 }
 
+export type GatsbyStateSlices = keyof IGatsbyState
```

Comment:
This gives you autocomplete for e.g. `['components', 'staticQueryComponents']` etc.